build_tools/cmake/iree_setup_toolchain.cmake

```diff
@@ -0,0 +1,34 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function(iree_append value)
+  foreach(variable ${ARGN})
+    set(${variable} "${${variable}} ${value}" PARENT_SCOPE)
+  endforeach(variable)
+endfunction()
+
+if(IREE_ENABLE_LLD)
+  set(IREE_USE_LINKER "lld")
```

Comment:
Maybe a check similar to https://github.com/llvm/llvm-project/blob/760f45eacadbabf9634fb81d7ccaa16c269cf19e/llvm/cmake/modules/HandleLLVMOptions.cmake#L260-L265 should be introduced here?

Reply:
As `IREE_USE_LINKER`, which is set here, is used in the main CMakeLists, it might be necessary to use `PARTENT_SCOPE`, but that is not more than a guess.

Reply:
Done. I don't think we need `PARENT_SCOPE` here given this is not in a function. It will be included and directly set in the file including this.